worlds/stardew_valley/test/TestMods.py

```diff
@@ -10,7 +13,23 @@
             "Ayeisha - The Postal Worker (Custom NPC)"]
 
 
-# Stealing from OptionSet in order to actually use it for tests.
+def check_stray_mod_items(mod: str, tester: SVTestBase, multiworld: MultiWorld):
```

Comment:
This whole thing can be made much simpler lol.

Same as the friendsanity methods from last week, don't call this so many times. Just once.

Then do something like
for item in multiworld.get_items():
    if item.mod_name != ModName.Vanilla:
        self.assertEquals(item.mod_name, mod)
for location in multiworld.get_locations():
    if location.mod_name != ModName.Vanilla:
        self.assertEquals(location.mod_name, mod)

Reply:
You can take a look at the "exclude ginger island" tests I made, which have something quite similar to what we're trying to achieve here.

Basically, from the multiworld.get_items(), you get the actual created items, and you are correct, these do not hold all the fields you need. But these items have names, so they can be mapped back to the full list of our items that are created from the CSV, using that name, and then you can check any of our custom flags or groups, including which mod the item is for.

Reply:
I went with a similar design as suggested.  Let me know if you think its doing too many passes.